etc/migration/10.3-to-11.0/mapped-emails.js

```diff
@@ -0,0 +1,122 @@
+/*!
+ * Copyright 2014 Apereo Foundation (AF) Licensed under the
+ * Educational Community License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may
+ * obtain a copy of the License at
+ *
+ *     http://opensource.org/licenses/ECL-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an "AS IS"
+ * BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * This migration script will try to create a mapping between a user's email address and the user
+ * object in the Principals table. Before it does that however, the following checks will be made:
+ *   -  Ensure that each user has an email address
+ *   -  Ensure that each user's email address is valid
+ *   -  Ensure that an email address is only used by 1 user
+ *
+ * A CSV file with all errors will be created in the current working directory. If any of the above
+ * checks fail, you will have to manually address these. How you resolve these issues is up to you.
+ *
+ * Once you've addressed the raised issues, re-run the script to perform the actual migration
```

Comment:
My bad, I wrote this migration script before we dropped the requirement of 1 user per email address. I seem to have missed quite a few spots, where the comments aren't synced up.